pkg/ingress2httpproxy/ingress2httpproxy.go

```diff
@@ -0,0 +1,120 @@
+package ingress2httpproxy
+
+import (
+	"strings"
+
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"github.com/prometheus/common/log"
+	"github.com/sirupsen/logrus"
+	core "k8s.io/api/networking/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	unSupportedHosts = "unsupported-hosts"
+)
+
+// MutatorOutput contains the mutated output structures
+type MutatorOutput struct {
+	HTTPProxy contourv1.HTTPProxy
+}
+
+// Mutator contains common atttributes and the mutation input source structure
+type Mutator struct {
+	name   string
+	log    logrus.FieldLogger
+	input  core.Ingress
+	domain string
+}
+
+// NewMutator creates a new Mutator. Clients of this API should set a meaningful name that can be used
+// to easily identify the calling client.
+func NewMutator(name string, log logrus.FieldLogger, ingress core.Ingress, domain string) Mutator {
+	return Mutator{
+		name:   name,
+		log:    log,
+		input:  ingress,
+		domain: domain,
+	}
+}
+
+// Mutate converts a Ingress into HTTPProxy
+func (m *Mutator) Mutate() *MutatorOutput {
+	return &MutatorOutput{
+		HTTPProxy: m.buildHTTPProxy(),
+	}
+}
+
+// buildHTTPProxy takes ingress object as an input and returns  Contour HTTPProxy
+func (m *Mutator) buildHTTPProxy() contourv1.HTTPProxy {
+
+	var httpAnnotations = make(map[string]string)
+	hpTranslatedRoute, httpAnnotations := m.createRoute(m.input.Spec.Rules, m.log, httpAnnotations)
+	hp := contourv1.HTTPProxy{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "HTTPProxy",
+			APIVersion: "projectcontour.io/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:        m.input.ObjectMeta.Name,
+			Annotations: httpAnnotations,
+			Namespace:   m.input.ObjectMeta.Namespace,
+		},
+		Spec: contourv1.HTTPProxySpec{
+			Routes: hpTranslatedRoute,
+		},
+	}
+
+	httpProxyFqdn := m.input.Spec.Rules[0].Host
+	log.Warnf("[%s] No new wildcard DNS domain specified. This mutation will use original domain from Ingress Host %s.", m.name, httpProxyFqdn)
+	if m.domain != "" {
+		normalizedDomain := m.domain
+		// let's accept the domain starting with "*." or "."
```

Comment:
if only certain input formats are excepted, might be good validate/report and error. for example using `example.com` vs. `.example.com` results in different outputs

Reply:
added the additional test cases